### PR TITLE
Fix the running out of mpi-communicators bug

### PR DESCRIPTION
**Pull request by aradi**
Even a single little can make much. Hopefully, this fixes the running out of mpi-communicators bug when DFTB+ is called via the API.

### CODE REVIEW

**Review by bhourahine — APPROVED** [review]
GTM

**Comment by aradi** [comment]
Thx! To be on the safe side, I will wait for confirmation from the original reporter and merge it then.

**Comment by bhourahine** [comment]
@aradi after 3 weeks, worth just merging it?

**Comment by bhourahine** [comment]
#800 is this fix on top of the new source tree structure.

**Pull request closed without merging**
